src/components/IdentitiesSwitch.js

```diff
@@ -0,0 +1,310 @@
+					}
+					iconBgStyle={styles.i_arrowBg}
+					iconType="antdesign"
+					iconName="arrowright"
+					iconSize={18}
+					textStyle={fontStyles.t_regular}
+					style={styles.i_arrowStyle}
+				/>
+			</>
+		);
+	};
+
+	const renderCurrentIdentityCard = () => {
+		if (!currentIdentity) return;
+
+		const currentIdentityTitle = getIdentityName(currentIdentity, identities);
+
+		return (
+			<>
+				<ButtonIcon
+					title={currentIdentityTitle}
+					onPress={() =>
+						onIdentitySelectedAndNavigate(
+							currentIdentity,
+							'AccountNetworkChooser'
+						)
+					}
+					iconType="antdesign"
+					iconName="user"
+					iconSize={40}
+					style={{ paddingLeft: 16 }}
+					textStyle={fontStyles.h1}
+				/>
+				{renderIdentityOptions(currentIdentity)}
+				<Separator style={{ marginBottom: 0 }} />
+			</>
+		);
+	};
+
+	const renderSettings = () => {
+		return (
+			<>
+				<ButtonIcon
+					title="About"
+					onPress={() => closeModalAndNavigate('About')}
+					iconType="antdesign"
+					iconName="info"
+					iconSize={24}
+					textStyle={fontStyles.t_big}
+					style={styles.indentedButton}
+				/>
+				<ButtonIcon
+					title="Terms and Conditions"
+					onPress={() =>
+						closeModalAndNavigate('TermsAndConditions', {
+							disableButtons: true
+						})
+					}
+					iconBgStyle={styles.i_arrowBg}
+					iconType="antdesign"
+					iconName="arrowright"
+					iconSize={18}
+					textStyle={fontStyles.t_regular}
+					style={styles.i_arrowStyle}
+				/>
+				<ButtonIcon
+					title="Privacy Policy"
+					onPress={() => closeModalAndNavigate('PrivacyPolicy')}
+					iconBgStyle={styles.i_arrowBg}
+					iconType="antdesign"
+					iconName="arrowright"
+					iconSize={18}
+					textStyle={fontStyles.t_regular}
+					style={styles.i_arrowStyle}
+				/>
+			</>
+		);
+	};
+
+	const renderNonSelectedIdentity = ({ item, index }) => {
+		const identity = item;
+		const title = identity.name || `identity_${index.toString()}`;
+
+		return (
+			<ButtonIcon
+				dropdown={false}
+				renderDropdownElement={() => renderIdentityOptions(identity)}
+				title={title}
+				onPress={() =>
+					onIdentitySelectedAndNavigate(identity, 'AccountNetworkChooser')
+				}
+				iconType="antdesign"
+				iconName="user"
+				iconSize={24}
+				style={styles.indentedButton}
+				textStyle={fontStyles.h2}
+			/>
+		);
+	};
+
+	const renderIdentities = () => {
+		// if no identity or the only one we have is the selected one
+
+		if (!identities.length || (identities.length === 1 && currentIdentity))
+			return;
+
+		const identitiesToShow = currentIdentity
+			? identities.filter(
+					identity => identity.encryptedSeed !== currentIdentity.encryptedSeed
+			  )
+			: identities;
+
+		return (
+			<>
+				<ScrollView style={{ maxHeight: 180 }}>
+					<FlatList
+						data={identitiesToShow}
+						renderItem={renderNonSelectedIdentity}
+						keyExtractor={item => item.encryptedSeed}
+						style={{ paddingVertical: identities.length > 5 ? 8 : 0 }}
```

Comment:
This (and the same check `>5` 2 lines down) looks sketchy :eyes:
Why `5` ?

Reply:
Just talked with Sveta, 8 is the Height of the scroll container.
If there are more than 5 identities, the list will become scrollable.
there will be shadow to indicate that there's more and some other details.

 So I think it is decide by the smallest screen, just checked on iPhone SE. Also tried on my big Android screen, there are more margin, but I think it is OK for now. We could programmatically change that as well.

Reply:
It would be better to discuss it separately, I think that would need discussion multiple possibilities together with Sveta, I will open an issue later this afternoon.

Reply:
![IMG_0044](https://user-images.githubusercontent.com/6014309/69337398-0bf22100-0c61-11ea-9468-9645aa10f182.PNG)